foreman/data_refinery_foreman/surveyor/sra.py

```diff
@@ -345,220 +205,186 @@ def _apply_metadata_to_experiment(experiment: Experiment, metadata: dict):
-            # merge it into the sample's existing protocol_info.
-            protocol_info, is_updated = self.update_sample_protocol_info(
-                sample_object.protocol_info,
-                experiment_object.protocol_description,
-                experiment_object.source_url,
-            )
-            if is_updated:
-                sample_object.protocol_info = protocol_info
-                sample_object.save()
 
-            logger.debug(
-                "Sample %s already exists, skipping object creation.",
-                sample_accession_code,
-                experiment_accession_code=experiment_object.accession_code,
-                survey_job=self.survey_job.id,
-            )
-        except Sample.DoesNotExist:
-            sample_object = Sample()
-            sample_object.source_database = "SRA"
-            sample_object.accession_code = sample_accession_code
-            sample_object.organism = organism
+        return experiment
+
+    def _generate_sample(self, experiment: Experiment, sample_metadata: dict) -> Sample:
+        """Generates or updates a Sample for the provided Experiment and sample metadata."""
+
+        run_accession = sample_metadata["run_accession"]
+
+        # Figure out the Organism for this sample
+        organism_name = sample_metadata.pop("scientific_name", None)
+        if not organism_name:
+            logger.error("Could not discover organism type for run.", accession=run_accession)
+            return (None, None)  # This will cascade properly
 
-            sample_object.platform_name = metadata.get("platform_instrument_model", "UNKNOWN")
+        organism = Organism.get_object_for_name(organism_name)
+
+        sample_accession = sample_metadata.pop("run_accession")
+        sample, sample_created = Sample.objects.get_or_create(accession_code=sample_accession)
+
+        if sample_created:
+            sample.source_database = "SRA"
+            sample.organism = organism
+            sample.platform_name = sample_metadata.get("instrument_model", "UNKNOWN")
             # The platform_name is human readable and contains spaces,
             # accession codes shouldn't have spaces though:
-            sample_object.platform_accession_code = sample_object.platform_name.replace(" ", "")
-            sample_object.technology = "RNA-SEQ"
-            if (
-                "ILLUMINA" in sample_object.platform_name.upper()
-                or "NEXTSEQ" in sample_object.platform_name.upper()
-            ):
-                sample_object.manufacturer = "ILLUMINA"
-            elif "ION TORRENT" in sample_object.platform_name.upper():
-                sample_object.manufacturer = "ION_TORRENT"
+            sample.platform_accession_code = sample.platform_name.replace(" ", "")
+            sample.technology = "RNA-SEQ"
+
+            platform_name = sample.platform_name.upper()
+            if platform_name in ["ILLUMINA", "NEXTSEQ"]:
+                sample.manufacturer = "ILLUMINA"
```

Comment:
Noting the reversal of logic here: I was assuming the previous configuration (`"ILLUMINA" in sample_object.platform_name.upper()`)was being used as a simple regex to catch things like `ILLUMINA MISEQ` which would be missed by the changes here.

Reply:
Related: do we want to add the other Illumina platforms we are planning to support here (i.e. MiSeq)? I'm not sure how well this will catch all of them: I would have expected `ILLUMINA` to be sufficient, but the inclusion of `NEXTSEQ` says otherwise!

Reply:
@jashapiro - I expect this list is guiding the logic here: https://github.com/AlexsLemonade/refinebio/blob/7d87133d820973692fa6465ff09ede9395422521/config/supported_rnaseq_platforms.txt 

Illumina MiSeq is listed as `Illumina MiSeq`, whereas the NextSeq platforms are listed as `NextSeq <number>`. That list was compiled using the platforms returned by ENA (see: https://github.com/AlexsLemonade/refinebio/pull/3281 for methods).

Reply:
Based on that list we will also need to add `HISEQ` and reverting to the previous structure (`"ILLUMINA" in sample.platform_name.upper()`, etc.) or a version that does preserve partial matches will be required.

Reply:
I looked through RNA-Seq samples in the database, that we have previously processed and the only manufacturers that are NEXTSEQ are for GEO accessions. I am unsure if that is a problem. I think that check was unnecessary as we should only be surveying geo accessions in the geo surveyor. All of our SRA accessions have unknown, illumina, or ion_torrent.

Reply:
>>> sra = Sample.objects.filter(source_database="SRA")
>>> sra.distinct('manufacturer').values_list('manufacturer', flat=True)
<QuerySet ['ILLUMINA', 'ION_TORRENT', 'UNKNOWN']>